### PR TITLE
Gets transfer fee by query instead of by config

### PR DESCRIPTION
**Pull request by milesz8**
https://github.com/osmosis-labs/osmosis-frontend/issues/982

- Uses axelar's AxelarQueryAPI to get fee's
- Removes unneeded config values

### CODE REVIEW

File: packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        environment,
+      });
+      const amount = Number(
+        new CoinPretty(
+          currency,
+          amountMinDenom === "" ? "0" : amountMinDenom
+        ).toCoin().amount
+      );
+      if (!isNaN(amount)) {
+        console.log("amount " + amount);
```
Comment (jonator) [inline packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts]: Please remove log.

File: packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        );
+      } else {
+        throw new Error("Requested fee amount is not a number.");
+      }
+    };
+    let timeout: NodeJS.Timeout | undefined;
+
+    setIsLoading(true);
+    // debounce query on user typing
+    timeout = setTimeout(() => {
```
Comment (jonator) [inline packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts:47]: Could you please create a general debounce hook in the /hooks/ folder? Note- you may be able to leverage a `debounce` package we already download into our bundle. It should be able to watch any given value and only run an effect once some debounce period has elapsed.

File: packages/web/integrations/axelar/transfer.tsx [hunk 3]
```diff
@@ -175,6 +174,20 @@ const AxelarTransfer: FunctionComponent<
     /** Amount, with decimals. e.g. 1.2 USDC */
     const amount = isWithdraw ? withdrawAmountConfig.amount : depositAmount;
 
+    console.log("Get transfer fee");
```
Comment (jonator) [inline packages/web/integrations/axelar/transfer.tsx]: Please remove all logs.

File: packages/web/integrations/axelar/transfer.tsx [hunk 4]
```diff
@@ -175,6 +174,20 @@ const AxelarTransfer: FunctionComponent<
...
+
+    const { transferFee: transferFeeMinAmount } = useTransferFeeQuery(
+      sourceChain,
+      destChain,
+      originCurrency.coinMinimalDenom,
+      amount,
+      originCurrency,
+      isTestNet ? Environment.TESTNET : Environment.MAINNET,
+      1000
```
Comment (jonator) [inline packages/web/integrations/axelar/transfer.tsx]: Is this too short of a debouce period? Perhaps 3 or 2 seconds?

File: packages/web/integrations/axelar/transfer.tsx [hunk 5]
```diff
@@ -175,6 +174,15 @@ const AxelarTransfer: FunctionComponent<
     /** Amount, with decimals. e.g. 1.2 USDC */
     const amount = isWithdraw ? withdrawAmountConfig.amount : depositAmount;
 
+    const { transferFee: transferFeeMinAmount } = useTransferFeeQuery(
+      sourceChain,
+      destChain,
```
Comment (jonator) [inline packages/web/integrations/axelar/transfer.tsx:179]: I think we want to leave this variable named `transferFee`. The min amount is used to disable UI if the amount isn't at least that. It's related but not the same.

**Review by jonator — CHANGES_REQUESTED** [review]

File: packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+/** Fetches a new Transfer fee quote when either chain, the amount, or the currency changes.
+ *  `amountMinDenom` is from user input, assumes `=== ""` for no input, and therefore no query.
+ */
+export function useTransferFeeQuery(
+  sourceChain: string,
+  destChain: string,
+  tokenMinDenom: string,
+  amountMinDenom: string,
+  currency: AppCurrency,
+  environment = Environment.MAINNET,
```
Comment (JeremyParish69) [inline packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts:15]: should we also allow for the TESTNET environment? The values are not equal, and the TESTNET values appear to be completely erased.

Comment (jonator) [reply packages/web/integrations/axelar/hooks/use-transfer-fee-query.ts:15]: If we pass in Environment.TESTNET we should have the testnet values

https://github.com/milesz8/osmosis-frontend/blob/5443d2e5d3b525c5309cb4d926dd55e0e3d2c3fd/packages/web/integrations/axelar/transfer.tsx#L183

File: packages/web/integrations/axelar/transfer.tsx (hunk 5 above)
Comment (milesz8) [reply packages/web/integrations/axelar/transfer.tsx:179]: okay

**Review by jonator — CHANGES_REQUESTED** [review]
It appears there's a build error (type error)

**Pull request merged**